src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -132,9 +132,13 @@ public async Task InvokeAsync(HttpContext context)
                 }
                 finally
                 {
-                    transaction.Finish(
-                        GetSpanStatusFromCode(context.Response.StatusCode)
-                    );
+                    transaction.Name = context.GetTransactionName(out var routeTemplateExists);
+                    if (routeTemplateExists || _options.TraceRequestsWithoutRouting)
```

Comment:
Sounds like the we should use ` _options.TraceRequestsWithoutRouting` when doing the sampling decision and mark the transaction as unsampled. Because this means the transaction will be created, outgoing HTTP requests will propagate it, but at the end we won't capture it here.

Reply:
@bruno-garcia We cannot, because we don't have the name of the route until after the request

Reply:
Discussed here: https://github.com/getsentry/sentry-dotnet/issues/716
We likely need a new middleware that will be added after the current one (once a route template exists). There we can start a transaction. but the error tracking part stays where it is.